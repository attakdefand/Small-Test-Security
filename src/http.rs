use axum::{routing::get, Router};
use serde::Serialize;

#[derive(Serialize)]
struct Health { status: &'static str, version: &'static str }

pub fn app() -> Router {
    Router::new().route("/health", get(|| async {
        axum::Json(Health { status: "ok", version: "v1" })
    }))
}
