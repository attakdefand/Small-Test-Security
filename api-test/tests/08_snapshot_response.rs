use axum::{Router, routing::get, body::Body};
use tower::util::ServiceExt;
use axum::http::Request;
use insta::assert_json_snapshot;

#[tokio::test]
async fn markets_json_stable() {
    let app = Router::new().route("/markets", get(|| async { axum::Json(serde_json::json!({"items":["BTC-USD","ETH-USD"]})) }));
    let res = app
        .oneshot(Request::builder().uri("/markets").body(Body::empty()).unwrap())
        .await
        .unwrap();
    let body = axum::body::to_bytes(res.into_body(), usize::MAX).await.unwrap();
    let val: serde_json::Value = serde_json::from_slice(&body).unwrap();
    assert_json_snapshot!(val);
}