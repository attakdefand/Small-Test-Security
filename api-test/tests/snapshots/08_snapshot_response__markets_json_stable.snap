---
source: api-test/tests/08_snapshot_response.rs
expression: val
---
{
  "items": [
    "BTC-USD",
    "ETH-USD"
  ]
}
